### PR TITLE
Fix incorrect range on maven property definition link

### LINKED ISSUE #107: Definition range for maven properties is incorrect
Doing a ctrl+click on a maven property will create a link for the entire node's text when in reality only the maven property should get a link.
image

### PR DESCRIPTION
**Pull request by AObuchow**
Fix #107

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

### CODE REVIEW

File: lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenDefinitionParticipant.java [hunk 1]
```diff
@@ -124,7 +126,22 @@ private LocationLink findMavenPropertyLocation(IDefinitionRequest request) {
 		if (propertyDeclaration == null) {
 			return null;
 		}
-		return toLocation(childProj.getFile(), propertyDeclaration, request.getNode());		
+		
+
+		String mavenPropertyText = "${" + mavenProperty + "}";
+		Position propertyStart;
+		try {
+			propertyStart = new Position(request.getPosition().getLine(),
```
Comment (AObuchow) [inline lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenDefinitionParticipant.java:134]: This is quite messy right now..

**Comment by AObuchow** [comment]
This still doesn't cover the case where two maven properties are use in the same tag's text. The second maven property won't have any definition link.

**Comment by mickaelistria** [comment]
I don't fully get the case you're mentioning. Can you please give an example?
In any case,as long as the result is better than current state, feel free to merge, even if there are some possible glitches (as long as those glitches are not regression).

**Comment by AObuchow** [comment]
@mickaelistria the case is trying to ctrl+click on "bar" in the following pom:
```XML
<project xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
	<modelVersion>4.0.0</modelVersion>

	<groupId>org.test</groupId>
	<artifactId>test</artifactId>
	<version>0.0.1-SNAPSHOT</version>
	<packaging>pom</packaging>

	<properties>
		<foo>$</foo>
		<bar>test</bar>
	</properties>
	<contributors>
		<contributor>
			<name>${foo}   ${bar}</name>
		</contributor>
	</contributors>
</project>
```

I'll improve this patch a bit, see if I can fix the described case and if not, merge and make a new issue for it.

**Pull request merged**